packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```
@@ -171,7 +177,11 @@ function act(callback: () => Thenable) {
 
     // flush effects until none remain, and cleanup
     try {
-      flushWork();
+      if (actingUpdatesScopeDepth === 1) {
```

Nit:
```suggestion
      if (actingUpdatesScopeDepth <= 1) {
```

---

really? it shouldn't do that if it's 0 (and it ideally should never be 0 here). going to push back on this.

---

tbh, it should probably throw an error if it ever gets there